tests/test_other.py

```diff
@@ -8813,6 +8813,65 @@ def test_main_module_without_main(self):
     run_process([PYTHON, EMCC, 'src.c', '--pre-js', 'pre.js', '-s', 'MAIN_MODULE=2'])
     self.assertContained('bar', run_js('a.out.js'))
 
+  def test_dynamicLibraries_i64(self):
+    create_test_file('pre.js', r'''
+if (typeof(Module) === "undefined") Module = {};
+Module.dynamicLibraries = ['side.wasm'];
+''')
+    create_test_file('side.hpp', r'''
+#include <emscripten.h>
+#include <cstdint>
+EMSCRIPTEN_KEEPALIVE int32_t function_ret_32(int32_t i, int32_t j, int32_t k);
+EMSCRIPTEN_KEEPALIVE int64_t function_ret_64(int32_t i, int32_t j, int32_t k);
+''')
+    create_test_file('side.cpp', r'''
+#include "side.hpp"
+int32_t function_ret_32(int32_t i, int32_t j, int32_t k) {
+    return 32;
+}
+int64_t function_ret_64(int32_t i, int32_t j, int32_t k) {
+    return 64;
+}
+''')
+    create_test_file('main.cpp', r'''
+#include <cstdio>
+#include <cinttypes>
+#include "side.hpp"
+
+typedef int32_t (*fp_type_32)(int32_t, int32_t, int32_t);
+typedef int64_t (*fp_type_64)(int32_t, int32_t, int32_t);
+
+int32_t internal_function_ret_32(int32_t i, int32_t j, int32_t k) {
+    return 32;
+}
+int64_t internal_function_ret_64(int32_t i, int32_t j, int32_t k) {
+    return 64;
+}
+
+int main() {
+    fp_type_32 fp32_internal = &internal_function_ret_32;
+    fp_type_32 fp32_external = &function_ret_32;
+    fp_type_64 fp64_external = &function_ret_64;
+    fp_type_64 fp64_internal = &internal_function_ret_64;
+    int32_t ires32 = fp32_internal(0,0,0);
+    printf("res32 - internal %d\n",ires32);
+    int32_t eres32 = fp32_external(0,0,0);
+    printf("res32 - external %d\n",eres32);
+
+    int64_t ires64 = fp64_internal(0,0,0);
+    printf("res64 - internal %" PRId64 "\n",ires64);
+    int64_t eres64 = fp64_external(0,0,0);
+    printf("res64 - external %" PRId64 "\n",eres64);
+    return 0;
+}
+''')
+    run_process([PYTHON, EMCC, 'side.cpp', '-s', 'SIDE_MODULE=2', '-o', 'side.wasm'])
+    run_process([PYTHON, EMCC, 'main.cpp', '-s', 'MAIN_MODULE=2', '--pre-js', 'pre.js'])
+    self.assertContained('''res32 - internal 32
+res32 - external 32
+res64 - internal 64
+res64 - external 64''', run_js('a.out.js'))
```

Comment:
Can you not use the same helper function that other `test_dylink_*` function in this file use?

Can we reduce this test to something smaller that just tests for the exact problem?  Why both to call the functions internally for example?

Reply:
I actually thought it might be good to write a new test here, as it uses `Module.dynamicLibraries`, and we have no coverage for that aside from a few in the `browser` suite. So it works differently than the existing dylink tests in `core`, and can't use the same infra.

Reply:
How does that work differently?  I thought RUNTIME_LINKED_LIBS just ends up setting dynamicLibraries, no?

If you want to write a specific test for modifying dynamicLibraries I don't see that as related to fixing this i64 issue.

Reply:
I agree it's separate, but it was easier to do both at once.

If you want I can rewrite this test from scratch?

Reply:
I rewrote the test in core as you preferred. It works (i.e. it fails before the PR, passes with it) - not sure what I did wrong before.